lib/terminal/terminal_interpreter.dart

```diff
@@ -0,0 +1,482 @@
+        break;
+      default:
+    }
+  }
+
+  List<int>? _tryParseAnsiBuffer(String terminatingCharacter) {
+    String value = ansiBuffer.toString();
+    List<int> res = [];
+
+    int num = 0;
+
+    for (int i = 0; i < value.length; i++) {
+      if (value[i] == ';') {
+        res.add((num / 10).toInt());
+        num = 0;
+        continue;
+      }
+
+      if (!_canConvertToNumber(value[i])) {
+        String remainingString = i + 1 < value.length
+            ? value.substring(i + 1)
+            : '';
+
+        for (int j = 0; j < remainingString.length; j++) {
+          if (currentColumn > columns - 1 || currentLine > lines - 1) break;
+          writeToBufferState(remainingString[j]);
+          currentColumn++;
+        }
+
+        if (currentColumn <= columns - 1 && currentLine <= lines - 1) {
+          bufferState[currentLine][currentColumn].character =
+              terminatingCharacter;
+        }
+
+        return null;
+      } else {
+        num = (num + int.parse(value[i])) * 10;
+      }
+    }
+
+    // any remaining unprocessed number
+    if (num > 0) res.add((num / 10).toInt());
+
+    return res;
+  }
+
+  void _eraseFromCursorToEnd() {
+    for (int i = currentColumn; i < columns; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  void _eraseFromBeginningToCursor() {
+    for (int i = 0; i <= currentColumn; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  void _eraseEntireLine() {
+    for (int i = 0; i < columns; i++) {
+      bufferState[currentLine][i].character = ' ';
+      bufferState[currentLine][i].backgroundColor = _backgroundColor;
+      bufferState[currentLine][i].foregroundColor = _foregroundColor;
+    }
+  }
+
+  String _matrixToString(List<List<String>> matrix) {
+    return matrix.map((row) => row.join()).join('\n');
+  }
+
+  String _formatMatrixToString(List<List<String>> matrix) {
+    final padded = _padMatrix(matrix);
+    return padded.map((row) => row.join()).join('\n');
+  }
+
+  List<List<String>> _padMatrix(List<List<String>> matrix) {
+    return matrix
+        .map((row) => row.map((cell) => cell.padRight(8)).toList())
+        .toList();
+  }
+
+  bool _canConvertToNumber(String char) {
+    return int.tryParse(char) != null;
+  }
+
+  void _clearStyles() {
+    _foregroundColor = null;
+    _backgroundColor = null;
+    _fontStyles.clear();
+  }
+
+  bool _validRgbColor(int r, int g, int b) {
+    bool validRed = 0 <= r && r < 256;
+    bool validGreen = 0 <= g && g < 256;
+    bool validBlue = 0 <= b && b < 256;
+
+    return validRed && validGreen && validBlue;
+  }
+
+  AnsiColorType _colorFromCode(int code) {
+    final mapping = <int, Colors>{
```

Comment:
Can this be a static const field?

Also I'd probably make a factory constructor `Color.fromCode(int code)` and move the const map into `Color`